src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -98,4 +104,22 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     darCollectionDAO.updateDarCollection(collectionId, user.getDacUserId(), new Date());
     return getByCollectionId(collectionId);
   }
+
+  //Writing this method here so it can be testable and reused for collection resources
+  //This method will be expanded to account for dataset binning once roles come into play
+  //For now it's functionality is simple, fetch the datasets and attach it to the collection
+  //DTO class is used to to it's dataUse attribute which will be needed for binning functionality (to be added soon)
+  public void addDatasetsToCollections(List<DarCollection> collections) {
+    for (DarCollection collection : collections) {
+      List<Integer> datasetIds = new ArrayList<Integer>();
+      for (DataAccessRequest dar : collection.getDars()) {
+        datasetIds.add(dar.getData().getDatasetIds().get(0));
+      }
+      List<Integer> distinctIds = datasetIds.stream().distinct().collect(Collectors.toList());
+      // Dataset call is being done seperatly to keep collection queries and collection mappers/reducers from getting messy
+      Set<DataSet> datasets = datasetDAO.findDatasetWithDataUseByIdList(distinctIds);
+      collection.setDatasets(datasets);
```

Comment:
I'm seeing odd results here and am not really sure why ... we should have distinct datasets in the set. I'm not seeing that though, I'm seeing cartesian products. Here's one example. Note the same dataset listed 8 times.
```
  {
    "darCollectionId": 258,
    "darCode": "DAR-288",
    "createDate": 1607447580992,
    "createUserId": 3351,
    "updateDate": null,
    "updateUserId": null,
    "dars": [
      {
        "id": 395,
        "referenceId": "e8b55bb4-2700-408c-9e45-1b08f1d296a4",
        "collectionId": 258,
        "data": {
          "referenceId": "e8b55bb4-2700-408c-9e45-1b08f1d296a4",
          "investigator": "Gregory Rushton",
          "institution": "Broad",
          "department": "DSP",
          "address1": "75 Ames St",
          "city": "Cambridge",
          "state": "Massachusetts",
          "country": "United States",
          "projectTitle": "Manual DS Insert Test",
          "checkCollaborator": false,
          "researcher": "Gregory Rushton",
          "isThePi": "true",
          "academicEmail": "grushton@broadinstitute.org",
          "rus": "Manual DS Insert Test",
          "nonTechRus": "Manual DS Insert Test",
          "diseases": false,
          "methods": false,
          "controls": false,
          "population": false,
          "other": false,
          "ontologies": [],
          "forProfit": false,
          "oneGender": false,
          "pediatric": false,
          "illegalBehavior": false,
          "addiction": false,
          "sexualDiseases": false,
          "stigmatizedDiseases": false,
          "vulnerablePopulation": false,
          "populationMigration": false,
          "psychiatricTraits": false,
          "notHealth": false,
          "hmb": true,
          "poa": false,
          "datasets": [
            {
              "key": "451",
              "value": "451",
              "label": "Synovial_Sarcoma-Suva_Riggi_Regev_V2"
            }
          ],
          "darCode": "DAR-288",
          "createDate": 1607447695236,
          "sortDate": 1607552371614,
          "datasetIds": [
            451
          ],
          "datasetDetail": [],
          "anvilUse": true,
          "cloudUse": false,
          "localUse": false,
          "itDirector": "IT Director",
          "signingOfficial": "Signing Official",
          "collaborationLetterLocation": "252e27c9-01db-47b6-8981-040b99f58f25",
          "collaborationLetterName": "single-cell_rna-seq_data_obtained_from_synovial_sarcoma_tumors.tsv",
          "labCollaborators": [],
          "internalCollaborators": [],
          "externalCollaborators": [],
          "pubAcknowledgement": false,
          "dsacknowledgement": false,
          "gsoacknowledgement": false
        },
        "draft": false,
        "userId": 3351,
        "createDate": 1607447580992,
        "sortDate": 1607552371623,
        "submissionDate": 1607447695236,
        "updateDate": 1607552371623
      }
    ],
    "datasets": [
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      },
      {
        "dataSetId": 451,
        "name": "Synovial_Sarcoma-Suva_Riggi_Regev_V2",
        "createDate": 1607385600000,
        "createUserId": 3351,
        "updateDate": 1607446488757,
        "updateUserId": 3351,
        "active": true,
        "alias": 86,
        "dataUse": {
          "hmbResearch": true,
          "populationOriginsAncestry": true,
          "commercialUse": false,
          "pediatric": true,
          "collaboratorRequired": true
        },
        "datasetIdentifier": "DUOS-000086"
      }
    ]
  }
```

Reply:
The root of the duplicate datasets is in `DataSetMapper` ... the older mapping classes do not do any kind of work to reduce the rows into single objects. Using a `RowReducer` implementation should fix this problem.